### PR TITLE
112 added option to specify commit interval in config file

### PR DESCRIPTION
**Pull request by viveksachdeva**
Issue Id:
https://github.com/TWCable/grabbit/issues/112

Changes:
Added Option to specify commit interval in config file. If not specified, defaults to 100.

@sagarsane @jdigger @jbornemann @jazeren1

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/TWCable/grabbit/pull/115" target="_blank">ReviewNinja

**Comment by jdigger** [comment]
It doesn't look like it's specified at the job/path level so that nodes under `/etc` can be treated differently than those under `/content/dam`. Does the way the Spring Batch configuration work not allow doing so?

**Comment by sagarsane** [comment]
@viveksachdeva could you also see if you could add / update `GrabbitConfigurationSpec` to account for this new feature?

**Comment by viveksachdeva** [comment]
@jdigger : It can be done at job level and if not present there, can be defaulted to transaction level..

**Comment by viveksachdeva** [comment]
@sagarsane : Sure.. will add those

**Comment by viveksachdeva** [comment]
@sagarsane @jdigger : I have updated code to pick batch size from pathConfiguration. So the priority order is:
- path configuration
- transaction
- default(100)
Test cases have also been updated..

**Comment by viveksachdeva** [comment]
@sagarsane @jdigger @masroormohammed @jazeren1 @jbornemann 

This PR is ready for review..

**Comment by jbornemann** [comment]
Thanks! I think this looks good!

File: grabbit/src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy [hunk 1]
```diff
@@ -152,6 +156,19 @@ class GrabbitConfiguration {
...
+            log.debug "Input doesn't contain ${key} for a boolean value. Will default to null"
+        }
+        Integer integerValue
+        try {
+            integerValue = configMap.get(key) as Integer
+        }
+        catch (NumberFormatException numberFormatException){
+            log.debug "${key} is not a number"
+        }
+        return integerValue
```
Comment (jdigger) [inline grabbit/src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy]: What does this return if `config.get(key)` contains "fooble"?

Comment (jdigger) [inline grabbit/src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy]: !resolve

**Comment by jdigger** [comment]
Other than the return value ambiguity, it looks good.

File: grabbit/src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy (hunk 1 above)
Comment (viveksachdeva) [inline grabbit/src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy]: !resolved

**Comment by viveksachdeva** [comment]
@jdigger : Added default value 0 in case value is not a number..

**Comment by viveksachdeva** [comment]
For testing:
With following sample config:
    "serverUsername" : "admin",
            "serverPassword" : "admin",
            "serverHost" : "localhost",
            "serverPort" : "4503",
            "deltaContent" : false,
            "batchSize"  : 50,
            "pathConfigurations" :  [
                {
                    "path" : "/content/a/b",
                    "batchSize"  : 150,
                },
                {
                    "path" : "/content/xyz",
                }
            ]
        }

there should be 2 jobs in the transaction:
- The one with path "/content/a/b" should have batch size 150
- The one with path "/content/xyz" should have batch size 50

If batchSize is not specified anywhere in the config, it should be 100.

**Comment by Shashi-Bhushan** [comment]
:+1: Testing 

Tested with following configurations
**{
	"serverUsername" : "admin",
	"serverPassword" : "admin",
	"serverHost" : "localhost",
	"serverPort" : "7503",
	"deltaContent" : false,
	"pathConfigurations" : [
		{
			"path" : "/content/usergenerated",
			"batchSize" : 150,
		},
		{
			"path" : "/content/campaigns"
		}
	]
}**

And

**{
	"serverUsername" : "admin",
	"serverPassword" : "admin",
	"serverHost" : "localhost",
	"serverPort" : "7503",
	"deltaContent" : false,
	"batchSize" : 50,
	"pathConfigurations" : [
		{
			"path" : "/content/usergenerated",
			"batchSize" : 150,
		},
		{
			"path" : "/content/campaigns"
		}
	]
}**

**Comment by jdigger** [comment]
What does a batch size of 0 do? (Or, for that matter, any non-positive number.)

**Comment by viveksachdeva** [comment]
@jdigger : Negative value of batch size was not handled before.. I have pushed a commit for the same.. Negative numbers and 0 are now ignored.. The next config in the order of priority is picked.. So if job level config is negative, the one at transaction level would be used and that would fallback to default batch size of 100..

**Comment by jdigger** [comment]
Perfect. +1

**Comment by jdigger** [comment]
@viveksachdeva Please squash that last commit, then I'll merge it in.

**Comment by jbornemann** [comment]
@viveksachdeva I think this looks good! I tested, and didn't see any issues!

I do think we will want to update the README to reflect this new configuration option

**Comment by viveksachdeva** [comment]
@jdigger , @jbornemann : Done.. 👍

**Comment by sagarsane** [comment]
👍

**Pull request merged**